Index: E:/Gabi/Lucru/Issue 112737/iText_src/src/core/com/lowagie/text/pdf/PdfGraphics2D.java
===================================================================
--- E:/Gabi/Lucru/Issue 112737/iText_src/src/core/com/lowagie/text/pdf/PdfGraphics2D.java	(revision 3115)
+++ E:/Gabi/Lucru/Issue 112737/iText_src/src/core/com/lowagie/text/pdf/PdfGraphics2D.java	(working copy)
@@ -436,14 +436,20 @@
             double width = 0;
             if (font.getSize2D() > 0) {
                 float scale = 1000 / font.getSize2D();
-                width = font.deriveFont(AffineTransform.getScaleInstance(scale, scale)).getStringBounds(s, getFontRenderContext()).getWidth() / scale;
+				Font derivedFont = font.deriveFont(AffineTransform.getScaleInstance(scale, scale));
+                width = derivedFont.getStringBounds(s, getFontRenderContext()).getWidth();
+				if (derivedFont.isTransformed())
+					width /= scale;
             }
             // if the hyperlink flag is set add an action to the text
             Object url = getRenderingHint(HyperLinkKey.KEY_INSTANCE);
             if (url != null && !url.equals(HyperLinkKey.VALUE_HYPERLINKKEY_OFF))
             {
             	float scale = 1000 / font.getSize2D();
-            	double height = font.deriveFont(AffineTransform.getScaleInstance(scale, scale)).getStringBounds(s, getFontRenderContext()).getHeight() / scale;
+				Font derivedFont = font.deriveFont(AffineTransform.getScaleInstance(scale, scale));
+            	double height = derivedFont.getStringBounds(s, getFontRenderContext()).getHeight();
+				if (derivedFont.isTransformed())
+					height /= scale;
                 double leftX = cb.getXTLM();
                 double leftY = cb.getYTLM();
                 PdfAction action = new  PdfAction(url.toString());
